packages/material-ui/src/TextField/TextField.d.ts

```diff
@@ -48,7 +48,7 @@ export interface BaseTextFieldProps
   /**
    * Props applied to the [`FormHelperText`](/api/form-helper-text/) element.
    */
-  FormHelperTextProps?: Partial<FormHelperTextProps>;
+  FormHelperTextProps?: FormHelperTextProps<React.ElementType, { component: React.ElementType }>;
```

Comment:
If this approach work, would it need we need to update all the other cases? For instance:
https://github.com/mui-org/material-ui/blob/b17e6856c33e96e1b6c5c5dfac327127cf4be705/packages/material-ui/src/Dialog/Dialog.d.ts#L92 once we migrate Paper to the overridable pattern? cc @eps1lon